test/Microsoft.AspNetCore.Server.KestrelTests/TestHelpers/MockLibuv.cs

```diff
@@ -84,14 +83,14 @@ unsafe public MockLibuv()
                             // Ensure any subsequent calls to uv_async_send
                             // create a new _onPostTcs to be completed.
                             _completedOnPostTcs = true;
-                            onPostTcs = _onPostTcs;
+
+                            // Calling TrySetResult outside the lock to avoid deadlock
```

Comment:
I think the comment is still accurate since the actually call to `TrySetResult` is made without the lock. I added to the end of the comment to indicate that we don't want to block the run loop.